advanced_source/cpp_cuda_graphs/mnist.cpp

```diff
@@ -0,0 +1,378 @@
+    return 1;
+  }
+  torch::Device device(device_type);
+
+  Net model;
+  model.to(device);
+
+  auto train_dataset =
+      torch::data::datasets::MNIST(kDataRoot)
+          .map(torch::data::transforms::Normalize<>(0.1307, 0.3081))
+          .map(torch::data::transforms::Stack<>());
+  const size_t train_dataset_size = train_dataset.size().value();
+  auto train_loader =
+      torch::data::make_data_loader<torch::data::samplers::SequentialSampler>(
+          std::move(train_dataset), kTrainBatchSize);
+
+  auto test_dataset =
+      torch::data::datasets::MNIST(
+          kDataRoot, torch::data::datasets::MNIST::Mode::kTest)
+          .map(torch::data::transforms::Normalize<>(0.1307, 0.3081))
+          .map(torch::data::transforms::Stack<>());
+  const size_t test_dataset_size = test_dataset.size().value();
+  auto test_loader =
+      torch::data::make_data_loader(std::move(test_dataset), kTestBatchSize);
+
+  torch::optim::SGD optimizer(
+      model.parameters(), torch::optim::SGDOptions(0.01).momentum(0.5));
+
+  torch::TensorOptions FloatCUDA =
+      torch::TensorOptions(device).dtype(torch::kFloat);
+  torch::TensorOptions LongCUDA =
+      torch::TensorOptions(device).dtype(torch::kLong);
+
+  torch::Tensor train_data =
+      torch::zeros({kTrainBatchSize, 1, 28, 28}, FloatCUDA);
+  torch::Tensor train_targets = torch::zeros({kTrainBatchSize}, LongCUDA);
+  torch::Tensor train_output = torch::zeros({1}, FloatCUDA);
+  torch::Tensor train_loss = torch::zeros({1}, FloatCUDA);
+
+  torch::Tensor test_data =
+      torch::zeros({kTestBatchSize, 1, 28, 28}, FloatCUDA);
+  torch::Tensor test_targets = torch::zeros({kTestBatchSize}, LongCUDA);
+  torch::Tensor test_output = torch::zeros({1}, FloatCUDA);
+  torch::Tensor test_loss = torch::zeros({1}, FloatCUDA);
+  torch::Tensor test_total_loss = torch::zeros({1}, FloatCUDA);
+  torch::Tensor test_total_correct = torch::zeros({1}, LongCUDA);
+
+  at::cuda::CUDAGraph train_graph;
+  at::cuda::CUDAGraph test_graph;
+
+  capture_train_graph(
+      model,
+      optimizer,
+      train_data,
+      train_targets,
+      train_output,
+      train_loss,
+      train_graph);
+
+  capture_test_graph(
+      model,
+      test_data,
+      test_targets,
+      test_output,
+      test_loss,
+      test_total_loss,
+      test_total_correct,
+      test_graph);
+
+  for (size_t epoch = 1; epoch <= kNumberOfEpochs; ++epoch) {
+    train(
+        epoch,
+        model,
+        device,
+        *train_loader,
+        optimizer,
+        train_dataset_size,
+        train_data,
+        train_targets,
+        train_output,
+        train_loss,
+        train_graph,
+        use_train_graph);
+    test(
+        model,
+        device,
+        *test_loader,
+        test_dataset_size,
+        test_data,
+        test_targets,
+        test_output,
+        test_loss,
+        test_total_loss,
+        test_total_correct,
+        test_graph,
+        use_test_graph);
+  }
```

Comment:
Add `return 0` and some message indicating that work was successful
```suggestion
  }
  std::cout << " Training/testing complete" << std::endl;
  return 0;
```